internal/util/util.go

```diff
@@ -0,0 +1,39 @@
+package util
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"fmt"
+	"io"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	kubeyaml "k8s.io/apimachinery/pkg/util/yaml"
+)
+
+// SplitYAMLToString splits a YAML file into strings. Returns list of yamls
+// found in the yaml. If an error occurs, returns objects that have been parsed so far too.
+func SplitYAMLToString(yamlData []byte) ([]string, error) {
```

Comment:
Looks like this covers https://github.com/numaproj-labs/numaplane/issues/128. Can you verify if it is the case for [raw files](https://github.com/numaproj-labs/numaplane/pull/119/files#diff-62540bd8eefb6e01f31efd127880279741d18ecc6845bc8f1f05a438db36a0b1R277) as well, thanks a lot!